src/guide/contributing-to-an-open-jdk-project.md

```diff
@@ -8,6 +8,30 @@ If you have a success story where Java solved your problem, or if you successful
 
 If you're in a position to choose what programming language to use in a project, in a tutorial, or in a class, you have the power to enlarge the Java community in a very direct way, and your colleagues or students will get an opportunity to learn one of the most used programming languages in the world.
 
+## Things to consider before proposing changes to OpenJDK code
+
+Every change in JDK code carries risk of changes in behavior which adversely affect applications. Generally we're looking to improve the functionality and capability and sometimes performance of the platform without that negative consequence. We do have many thousands of tests, but they're inevitably incomplete in coverage. So we need to ask ourselves whether each change is worthwhile - and some may not be no matter how well intentioned.
```

Comment:
> We do have many thousands of tests, but they're inevitably incomplete in coverage.

This sentence needs more explanation, I think. If you are changing code, ideally you would provide tests that test that behavior change if the current tests are inadequate. Does this sentence add anything to the main points in this paragraph - otherwise, maybe it should be removed.

Reply:
Agreed. Removed.